cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeMasterEnvironment.java

```diff
@@ -201,6 +212,36 @@ public MasterEnvironmentRunnable createRunnable(MasterEnvironmentRunnableContext
       .newInstance(context, this);
   }
 
+  @Override
+  public SparkConfig getSparkSubmitConfig() {
+    // Get k8s master path for spark submit
+    String master = getMasterPath();
+
+    // Create pod spec with cdap conf.
+    V1Pod v1Pod = new V1Pod();
+    v1Pod.setSpec(getPodSpecWithCConf(podInfo));
+
+    // Create spark template file. This file gets created in driver pod. We do not delete it because pod will get
+    // deleted at the end of job completion.
+    File templateFile = new File("podTemplate");
+    try (FileWriter writer = new FileWriter(templateFile)) {
+      writer.write(Yaml.dump(v1Pod));
+    } catch (IOException e) {
+      // should not happen
+      throw new RuntimeException("Exception while writing pod spec to temp file.", e);
+    }
+    additionalSparkConfs.put("spark.kubernetes.driver.podTemplateFile", templateFile.getAbsolutePath());
```

Comment:
Can `spark.kubernetes.driver.podTemplateFile` and other strings be refactored into private static fields or into a constants file?